### ISSUE TITLE
Implement Microsoft.Fsharp.Core.String.filter

### ISSUE THREAD

**Issue by jgrund** [fable-compiler/Fable#1132]
I was trying to filter characters out of a string and realized that `String.filter` is not implemented.

It should be able to be done in JS as something like:

```
export function filter(pred:(i: string) => string, x: string) {
  return x.split('').filter(pred).join('');
}
```

**Comment by alfonsogarciacaro** [comment]
Ups, I must have missed that! 😅 Could you please send a PR? It should be enough by adding the proposed function to String.ts, a test to StringTests.fs, and a line like the following [before this one](https://github.com/fable-compiler/Fable/blob/e1e369bdb33fcdc58a18c6b777fe3370fc85f245/src/dotnet/Fable.Compiler/Replacements.fs#L1038):

```fsharp
        | "filter" -> ccall i "String" "filter" i.args
```

**Comment by alfonsogarciacaro** [comment]
In exchange of that I've fixed #1131 ;)

**Comment by jgrund** [comment]
Fair enough :)
